actions/send-ci-cd-trace/src/index.ts

```diff
@@ -0,0 +1,154 @@
+import * as core from "@actions/core";
+import * as github from "@actions/github";
+import * as appInsights from "applicationinsights";
+import { DefaultAzureCredential } from "@azure/identity";
+
+async function run() {
+  try {
+    // Retrieve inputs
+    const connectionString = core.getInput("connection_string");
+    const token = core.getInput("repo_token");
+
+    if (!connectionString) {
+      throw new Error("Application Insights connection string is required.");
+    }
+
+    if (!token) {
+      throw new Error("GitHub token is required.");
+    }
+
+    // Set up Azure Application Insights with Azure Identity
+    const credential = new DefaultAzureCredential();
+    appInsights
+      .setup(connectionString)
+      .setAutoCollectRequests(true)
+      .setAutoCollectPerformance(true, false)
+      .setAutoCollectExceptions(true)
+      .setAutoCollectDependencies(true)
+      .setAutoCollectConsole(true, false)
+      .setAutoCollectPreAggregatedMetrics(true)
+      .setSendLiveMetrics(false)
+      .setInternalLogging(false, true)
+      .enableWebInstrumentation(false)
+      .start();
+
+    const client = appInsights.defaultClient;
+    client.config.aadTokenCredential = credential;
+
+    // Get GitHub context
+    const { repo, runId, job } = github.context;
+    const octokit = github.getOctokit(token);
+
+    // Fetch workflow run details from GitHub API
+    const getWorkflowRunResponse = await octokit.rest.actions.getWorkflowRun({
+      owner: repo.owner,
+      repo: repo.repo,
+      run_id: runId,
+    });
+
+    const workflowRun = getWorkflowRunResponse.data;
+
+    // Fetch jobs related to this workflow run
+    const jobsResponse = await octokit.rest.actions.listJobsForWorkflowRun({
+      owner: repo.owner,
+      repo: repo.repo,
+      run_id: runId,
+    });
+
+    const jobs = jobsResponse.data.jobs;
+    
+    // Check if the current job is the last job in the workflow
+    const isLastJob = jobs[jobs.length - 1]?.name === job;
+
+    // Collect workflow metadata
+    const workflowMetadata = {
+      app: repo.repo,
+      repository: repo.repo,
+      run_id: runId,
+      workflow_name: workflowRun.name,
+      status: workflowRun.status,
+      run_started_at: workflowRun.run_started_at,
+      created_at: workflowRun.created_at,
+      updated_at: workflowRun.updated_at,
+    };
+
+    // Track the workflow start trace
+    const workflowStartTime = new Date(workflowRun.run_started_at || workflowRun.created_at || 0);
+    client.trackEvent({
+      name: `Workflow: ${workflowRun.name}`,
+      properties: {
+        ...workflowMetadata,
+        startTime: workflowStartTime.toISOString(),
+      },
+    });
+
+    // Iterate over the jobs and track each job's information
+    for (const job of jobs) {
+      // Skip the job if it is not the last one
+      if (!isLastJob) {
+        continue;
+      }
```

Comment:
Are we only sending metrics if this action is used in the last job? If so we need to document the usage better I think. Because now I think the step will log `Trace data sent to Azure Monitor successfully.` even dough no trace data is sent?

If this is correct couldn't this check be placed outside the for loop as the `isLasJob` does not change it's value while the loop is running 🤔